services/terraformer/server/clusterBuilder/clusterBuilder.go

```diff
@@ -277,3 +323,39 @@ func getTplFile(clusterType pb.ClusterType) string {
 	}
 	return ""
 }
+
+// getCIDR function returns CIDR in IPv4 format, with position replaced by value
+// The function does not check if it is a valid CIDR/can be used in subnet spec
+func getCIDR(baseCIDR string, position int, existing map[string]struct{}) (string, error) {
+	_, ipNet, err := net.ParseCIDR(baseCIDR)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse a CIDR with base %s, position %d", baseCIDR, position)
+	}
+	ip := ipNet.IP
+	ones, _ := ipNet.Mask.Size()
+	var i int
+	for {
+		if i > 255 {
+			return "", fmt.Errorf("maximum number of IPs assigned")
+		}
+		ip[position] = byte(i)
+		if _, ok := existing[fmt.Sprintf("%s/%d", ip.String(), ones)]; ok {
+			// CIDR already assigned.
+			i++
+			continue
+		} else {
+			// CIDR does not exist yet, break.
+			return fmt.Sprintf("%s/%d", ip.String(), ones), nil
+		}
+	}
+}
+
+// copyCIDRsToMetadata copies CIDRs for subnets in VPCs for every nodepool.
+func copyCIDRsToMetadata(data *NodepoolsData) {
+	if data.Metadata == nil {
+		data.Metadata = make(map[string]any)
```

Comment:
why are we mapping to `any`? is there any structure that would be better suited to describe the intent of the map?

Reply:
No, as this map holds any data, we might need in the templates. Currently, it holds LB roles for LB templates and CIDRs for VPC subnets.